src/exa.c

```diff
@@ -272,14 +272,23 @@ unsigned TegraPixmapSize(TegraPixmapPtr pixmap)
     return TEGRA_ALIGN(size, TEGRA_EXA_OFFSET_ALIGN);
 }
 
+unsigned TegraEXAHeight(unsigned int height, unsigned int bpp)
+{
+    /* some of GR2D units operate with 16x16 (bytes) blocks */
+    if (bpp == 8 || bpp == 16 || bpp == 32)
+        return TEGRA_ALIGN(height, 16 / (bpp >> 3));
```

Comment:
Right. I thought this was only used for accelerated operations. And TBH, looking at the call-site, this would be much clearer if it was this way (and slightly faster, as it'd reuse the same comparison that happens a line after this call)...

Reply:
Yes, the height alignment is needed only for the accelerated operations. Could you please clarify what you're meaning by "this way"?

Reply:
"this was" = lift the `bpp == 8 || bpp == 16 || bpp == 32` condition out of `TegraEXAHeight()`. The function is a NOP unless that happens, and at least in TegraEXAAllocatePixmapData(), the exact same check happens right after. You'd probably want to put `bpp == 8 || bpp == 16 || bpp == 32` in a helper (Something like `bool TegraEXAAccelerated(int bpp) { return bpp == 8 || bpp == 16 || bpp == 32; }`?) so it reads a bit better also...